src/util/ConfigManager/ConfigManager.h

```diff
@@ -40,6 +41,60 @@ namespace ad_utility {
 Manages a bunch of `ConfigOption`s.
 */
 class ConfigManager {
+  /*
+  Our hash map always saves either a configuration option, or a sub manager
+  (another `ConfigManager`).
+
+  This class makes the handeling more explicit and allows for more information
+  to be saved alongside the configuration options and sub managers.
+  For example: The order, in which they were created.
+  */
+  class HashMapEntry {
+   public:
+    using Data = std::variant<ConfigOption, ConfigManager>;
+
+   private:
+    std::unique_ptr<Data> data_;
+
+    // Describes the order of initialization.
+    static inline size_t numberOfInstances_{0};
+    size_t initializationId_{numberOfInstances_++};
```

Comment:
If you are going the easy hacky way of making this a global static, make it `atomic<size_t>` for increased safety.

Reply:
Done, see [9d86c3d](https://github.com/ad-freiburg/qlever/pull/1153/commits/9d86c3ddbae62e2bea452201b9009fecf8b4317c).

How would the non hacky way look like? I thought of saving the date and time, but I'm not sure if those values would be precise enough.